nilearn/plotting/img_plotting.py

```
@@ -556,6 +556,158 @@ def plot_roi(roi_img, bg_img=MNI152TEMPLATE, cut_coords=None,
+
+    if hasattr(threshold, '__iter__') and not isinstance(threshold, _basestring):
+        if len(threshold) != n_maps:
+            raise TypeError('The list of values to threshold '
+                            'should be equal to number of maps')
+    else:
+        threshold = [threshold] * n_maps
+
+    for i, (map_img, color, thr) in enumerate(zip(iter_img(maps_img), color_list, threshold)):
+        # To threshold or choose the level of the contours
+        data = map_img.get_data()
+        if isinstance(thr, _basestring):
+            message = ("If 'threshold' is given as a string it "
+                       'should be a number followed by the percent sign, '
+                       'e.g. "25.3%"')
+            if not thr.endswith('%'):
+                raise ValueError(message)
+            try:
+                percentile = float(thr[:-1])
+            except ValueError as exc:
+                exc.args += (message, )
+                raise
+            thr = fast_abs_percentile(data, percentile) + 1e-1
+        elif not isinstance(thr, numbers.Real):
+            raise TypeError('Threshold must be a real value and you gave '
+                            'a %s.' % type(thr))
+        if view_type == 'auto':
+            if n_maps > 4:
+                view_type = 'contours'
+            else:
+                view_type = 'continuous'
+        elif view_type == 'contours':
+            display.add_contours(map_img, levels=[thr],
+                                 linewidths=linewidths,
+                                 colors=[color])
+            if filled:
+                # Append the lower boundary value as 0 for contour fillings
+                display.add_contours(map_img, levels=[thr, 0.],
+                                     linewidths=linewidths,
+                                     colors=[color[:3].tolist() + [alpha]],
+                                     linestyles='solid', filled=filled)
+        elif view_type == 'continuous':
```

You should raise an error if `view_type` is not an accepted value.